awx/main/management/commands/inventory_import.py

```diff
@@ -988,12 +910,28 @@ def handle(self, *args, **options):
                         self.inventory_update.status = 'running'
                         self.inventory_update.save()
 
-            # Load inventory from source.
-            self.all_group = load_inventory_source(self.source,
-                                                   self.group_filter_re,
-                                                   self.host_filter_re,
-                                                   self.exclude_empty_groups,
-                                                   self.is_custom)
+            source = self.get_source_absolute_path(self.source)
+
+            data = AnsibleInventoryLoader(source=source, is_custom=self.is_custom).load()
+
+            logger.debug('Finished loading from source: %s', source)
+            logger.info('Processing JSON output...')
+            inventory = MemInventory(
+                group_filter_re=self.group_filter_re, host_filter_re=self.host_filter_re)
+            inventory = dict_to_mem_data(data, inventory=inventory)
+
+            del data  # forget dict from import, could be large
+
+            logger.info('Loaded %d groups, %d hosts', len(inventory.all_group.all_groups),
+                        len(inventory.all_group.all_hosts))
+
+            # TODO: pass data, not inventory, into inventory saver, exclude groups there
```

Comment:
This still a `TODO`?

Reply:
The statement here isn't very clear, and for that reason, I could remove it. However, the MemInventory classes and similar ones: https://github.com/ansible/awx/blob/devel/awx/main/utils/mem_inventory.py could really stand to be refactored out. Back when we maintained our own logic for parsing .ini files, this code structure made a whole lot more sense. Now, these don't really offer a clarity benefit as opposed to keeping the data in dicts.